### PR TITLE
ITB-1010: Swagger changes for ITB-979

### PR DESCRIPTION
**Pull request by jack-durrant**
[ITB-1010](https://io1dev.atlassian.net/browse/ITB-1010)
[ITB-979](https://io1dev.atlassian.net/browse/ITB-979)

### CODE REVIEW

File: api-private.swagger.yaml [hunk 1]
```diff
@@ -208,14 +198,45 @@ paths:
...
+          in: path
+          type: string
+          description: The UUID fo the node to export.
+          pattern: '^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$'
+          required: true
+        - name: latest
+          in: query
+          type: boolean
+          description: Whether to use the latest revision of each page.
+          required: false
```
Comment (jmorahan) [inline api-private.swagger.yaml]: needs a default to specify what value is assumed if its omitted

File: api-private.swagger.yaml [hunk 2]
```diff
@@ -208,14 +198,45 @@ paths:
...
+      responses:
+        '200':
+          description: An existing PDF export job was found and returned to the client.
+          schema:
```
Comment (jmorahan) [inline api-private.swagger.yaml]: should be content / mime-type / schema eg:
```
content:
  application/json:
    schema:
      $ref: ...
```

File: api-private.swagger.yaml [hunk 3]
```diff
@@ -208,14 +198,45 @@ paths:
...
+            $ref: '#/definitions/pdf-export-job-schema'
+        '201':
+          description: A new PDF export job was created.
+          schema:
```
Comment (jmorahan) [inline api-private.swagger.yaml]: see above

File: api-private.swagger.yaml (hunk 2 above)
Comment (jack-durrant) [reply api-private.swagger.yaml]: Just copying how it's done elsewhere in the file. When I apply your suggested change, I get an error; see the screenshot below:

image

I've noticed that this particular file uses version 2 of the Swagger Specification, while the other file uses OAS version 3. Do you think that could be the cause of the error?

Comment (jmorahan) [reply api-private.swagger.yaml]: Ah! Yes, sorry I didn't realize this was still on Swagger 2. Let's leave it as you have it for now and update it to OAS 3 as a separate task.

File: api-private.swagger.yaml (hunk 1 above)
Comment (jack-durrant) [reply api-private.swagger.yaml]: Thanks; now defaults to `false`.

File: api-private.swagger.yaml (hunk 2 above)
Comment (jack-durrant) [reply api-private.swagger.yaml]: [ITB-1022](https://io1dev.atlassian.net/browse/ITB-1022) assigned to @jmorahan.

File: api-private.swagger.yaml [hunk 4]
```diff
@@ -208,14 +198,46 @@ paths:
           description: The JSON was malformed.
         '422':
           description: JSON schema validation failed.
+  '/pdf-export/create-job/{uuid}':
```
Comment (jmorahan) [inline api-private.swagger.yaml]: `content_id` for consistency :)

File: api-private.swagger.yaml [hunk 5]
```diff
@@ -208,14 +198,46 @@ paths:
...
+    post:
+      tags:
+        - pdf export
+      summary: Create a PDF export job.
+      parameters:
+        - name: uuid
+          in: path
+          type: string
+          description: The UUID fo the node to export.
```
Comment (jmorahan) [inline api-private.swagger.yaml]:
```suggestion
          description: The UUID of the node to export.
```

File: api-private.swagger.yaml (hunk 4 above)
Comment (jack-durrant) [reply api-private.swagger.yaml]: Thanks; done.

File: api-private.swagger.yaml [hunk 6]
```diff
@@ -208,14 +198,46 @@ paths:
           description: The JSON was malformed.
         '422':
           description: JSON schema validation failed.
+  '/pdf-export/create-job/{content_id}':
+    post:
+      tags:
+        - pdf export
+      summary: Create a PDF export job.
+      parameters:
+        - name: uuid
```
Comment (jmorahan) [inline api-private.swagger.yaml]:
```suggestion
        - name: content_id
```

**Pull request merged**